common/credentials/remoteconfig.go

```diff
@@ -420,72 +432,168 @@ func (rc *RemoteConfig) GetStorageType() (storage.Dest, error) {
 	}
 }
 
-func (rc *RemoteConfig) GetStorageCreds(typ storage.Dest) (*StorageCreds, error) {
-	switch typ {
+// GetStorageCreds initializes datastore info based on the configured storage type in Consul
+// FIXME: We're doing ourselves a disservice by forcing a user to pass in a *storage.Dest when we can handle this internally through rc
+func (rc *RemoteConfig) GetStorageCreds(typ *storage.Dest) (*StorageCreds, error) {
+	switch *typ {
 	case storage.Postgres:
 		return rc.getForPostgres()
 	case storage.FileSystem:
 		return rc.getForFilesystem()
 	default:
-		fmt.Println("shouldnoteverhappen")
-		return nil, nil
+		return nil, errors.New("Failed to get storage creds for Postgres or Filesystem. This shouldn't ever happen, but it did")
 	}
 }
 
+// getForPostgres Reads configuration from Consul
 func (rc *RemoteConfig) getForPostgres() (*StorageCreds, error) {
-	pairs, err := rc.Consul.GetKeyValues(common.PostgresCredLoc)
-	if err != nil {
-		return nil, errors.New("unable to get postgres creds from consul, err: " + err.Error())
+
+	// Credentials for Postgres may come from 2 mutually exclusive areas in Vault
+	// Either creds are static, and stored as a KV secret in vault
+	// Or creds are dynamic, and managed by vault's database secret engine
+
+	// The default behavior is to use static secrets
+
+	vaultConf, err := rc.Consul.GetKeyValues(common.VaultConf)
+	if len(vaultConf) == 0 || err != nil {
+		errorMsg := fmt.Sprintf("unable to get vault operating mode from consul")
+		if err != nil {
+			errorMsg = fmt.Sprintf("%s, err: %s", errorMsg, err.Error())
+		}
+		return &StorageCreds{}, errors.New(errorMsg)
+	}
+
+	//vaultRoleName := ""
+	vaultBackend := "kv" // The default backend is "kv" if left unconfigured in Consul
+	for _, vconf := range vaultConf {
+		switch vconf.Key {
+		case common.VaultDBSecretEngine:
+			switch string(vconf.Value) {
+			case "kv":
+				ocelog.Log().Debugf("Static Postgres creds")
+				break
+			case "database":
+				ocelog.Log().Debugf("Dynamic Postgres creds")
+				break
+			default:
+				return &StorageCreds{}, errors.New("Unsupported Vault DB Secret Engine")
+			}
+			vaultBackend = string(vconf.Value)
+			//case common.VaultRoleName:
+			//	vaultRoleName = string(vconf.Value)
+			//	break
+		}
 	}
+	//ocelog.Log().Debugf("vaultRoleName: %s\nvaultBackend: %s", vaultRoleName, vaultBackend)
+
 	storeConfig := &StorageCreds{}
-	for _, pair := range pairs {
+
+	kvconfig, err := rc.Consul.GetKeyValues(common.PostgresCredLoc)
+	if len(kvconfig) == 0 || err != nil {
+		errorMsg := fmt.Sprintf("unable to get postgres creds from consul")
+		if err != nil {
+			errorMsg = fmt.Sprintf("%s, err: %s", errorMsg, err.Error())
+		}
+		return &StorageCreds{}, errors.New(errorMsg)
```

Comment:
can just return nil because its a pointer returned
```suggestion
		return nil, errors.New(errorMsg)
```